code/modules/projectiles/guns/energy/modular.dm

```diff
@@ -79,12 +79,10 @@
 			gun_type = CHASSIS_MEDIUM
 			slot_flags = SLOT_BELT | SLOT_BACK
 			item_state = "energystun"
-			action_button_name = "Wield rifle"
```

Comment:
Missing `is_wieldable = TRUE` here and below? Though you will also need to make logic to actually add the button in (and remove it in `reset_vars`). You might want to subproc the initial switch as well into `updated_is_wieldable()` or something.